.gitlab-ci.yml

```diff
@@ -415,6 +430,24 @@ examples-contract-build:
       done
     - cargo +stable contract build --manifest-path ./integration-tests/upgradeable-contracts/set-code-hash/updated-incrementer/Cargo.toml
 
+# we only check a few basic contracts here. no need to check them all.
+examples-no-std-check:
+  stage:                           examples
+  <<:                              *docker-env
+  <<:                              *test-refs
+  variables:
+    RUSTC_BOOTSTRAP:               "1"
+  script:
+    - rustup component add rust-src --toolchain stable
+    - cargo contract -V
+    - for example in integration-tests/*/; do
+        if [ "$example" = "integration-tests/upgradeable-contracts/" ]; then continue; fi;
+        if [ "$example" = "integration-tests/lang-err-integration-tests/" ]; then continue; fi;
+        pushd $example &&
+        cargo +stable check --no-default-features --target $NO_STD_TARGET -Zbuild-std="core,alloc" &&
```

Comment:
This is still a lot of examples, wonder if we should just spot check a few instead (e.g `flipper`, `erc20`, etc.)

Reply:
We could but I don't think it causes any harm to just check all of them. Example checks are run in parallel and this is much faster than the rest anyways.